cl/recap/tasks.py

```diff
@@ -1419,3 +1428,71 @@ def mark_fq_status(fq, msg, status):
     if status == PROCESSING_STATUS.SUCCESSFUL:
         fq.date_completed = now()
     fq.save()
+
+
+@app.task(bind=True, max_retries=5, interval_start=5 * 60)
+def process_recap_email(self: Task, pk: int) -> Optional[int]:
+    start_time = now()
+    epq = EmailProcessingQueue.objects.get(pk=pk)
+    mark_pq_status(epq, "", PROCESSING_STATUS.IN_PROGRESS, "status_message")
+
+    message_id = epq.message_id
+    bucket = RecapEmailSESStorage()
+    body = bucket.open(message_id, "r").read()
+    report = S3NotificationEmail(map_cl_to_pacer_id(epq.court_id))
+    report._parse_text(body)
+    data = report.data
+
+    if (
+        data == {}
+        or data.get("docket_entries", None) is None
+        or len(data["docket_entries"]) == 0
+    ):
+        msg = "Not a valid notification email. No message content."
+        mark_pq_status(
+            epq, msg, PROCESSING_STATUS.INVALID_CONTENT, "status_message"
+        )
+        self.request.chain = None
+        return None
+
+    docket_entry = data["docket_entries"][0]
+    docket = find_docket_object(
+        epq.court_id, docket_entry["pacer_case_id"], data["docket_number"]
+    )
+
+    docket.add_recap_source()
+    update_docket_metadata(docket, data)
+
+    if not docket.pacer_case_id:
+        docket.pacer_case_id = docket_entry["pacer_case_id"]
+
+    docket.save()
+
+    # Add the HTML to the docket in case we need it someday.
+    pacer_file = PacerHtmlFiles(
+        content_object=docket, upload_type=UPLOAD_TYPE.SES_EMAIL
+    )
+    pacer_file.filepath.save(
+        "docket.txt",  # We only care about the ext w/UUIDFileSystemStorage
+        ContentFile(body),
+    )
+
+    rds_created, content_updated = add_docket_entries(
+        docket, data["docket_entries"]
+    )
+
+    if content_updated:
+        newly_enqueued = enqueue_docket_alert(docket.pk)
+        if newly_enqueued:
+            send_docket_alert(docket.pk, start_time)
+
+    return {
+        "docket_pk": docket.pk,
+        "content_updated": bool(rds_created or content_updated),
+    }
```

Comment:
I haven't verified that this covers the bases, but I can tell you grabbed it from elsewhere, so I assume so, but if you want me to review it more closely, let me know and I'm happy to do so.

Reply:
It is grabbed. This does not have a ton of original ideas from me. The big difference is retrieving the email and how we get information from Juriscraper.